versioned/storage/store/src/test/java/org/projectnessie/versioned/storage/versionstore/TestKeyRanges.java

```diff
@@ -0,0 +1,117 @@
+        .withMessageContaining("Either specify minKey+maxKey or prefixKey, but not both.");
+    soft.assertThatIllegalArgumentException()
+        .isThrownBy(() -> keyRanges(null, ContentKey.of("foo"), null, ContentKey.of("foo")))
+        .withMessageContaining("Either specify minKey+maxKey or prefixKey, but not both.");
+  }
+
+  @Test
+  public void pagingTokenSupersedesMinKey() {
+    String token =
+        pagingToken(copyFromUtf8(keyToStoreKey(ContentKey.of("foo")).rawString())).asString();
+
+    soft.assertThat(keyRanges(token, ContentKey.of("bar"), null, null))
+        .extracting(KeyRanges::beginStoreKey)
+        .isEqualTo(keyToStoreKey(ContentKey.of("foo")));
+
+    soft.assertThat(keyRanges(token, null, null, ContentKey.of("bar")))
+        .extracting(KeyRanges::beginStoreKey)
+        .isEqualTo(keyToStoreKey(ContentKey.of("foo")));
+  }
+
+  @Test
+  public void max() {
+    ContentKey bar = ContentKey.of("bar");
+    KeyRanges keyRange = keyRanges(null, null, bar, null);
+
+    soft.assertThat(keyRange.endStoreKey()).isGreaterThan(keyToStoreKey(bar));
+    soft.assertThat(keyRange.endStoreKey()).isGreaterThan(keyToStoreKeyVariant(bar, "Z"));
+    soft.assertThat(keyRange.endStoreKey()).isGreaterThan(keyToStoreKeyVariant(bar, "A"));
+
+    soft.assertThat(keyRange.endStoreKey())
+        .isLessThanOrEqualTo(keyToStoreKey(ContentKey.of("bar", "baz")));
+    soft.assertThat(keyRange.endStoreKey())
+        .isLessThanOrEqualTo(keyToStoreKey(ContentKey.of("bar", "0")));
+    soft.assertThat(keyRange.endStoreKey())
+        .isLessThanOrEqualTo(keyToStoreKey(ContentKey.of("barbaz")));
+    soft.assertThat(keyRange.endStoreKey())
+        .isLessThanOrEqualTo(keyToStoreKey(ContentKey.of("bar0")));
+
+    soft.assertThat(keyRange.endStoreKey())
+        .isLessThanOrEqualTo(keyToStoreKey(ContentKey.of("bar0")));
+
+    String storeKeyRawPrefix = MAIN_UNIVERSE + (char) 0 + "bar";
+
+    soft.assertThat(keyRange.endStoreKey()).isGreaterThan(keyFromString(storeKeyRawPrefix));
+    soft.assertThat(keyRange.endStoreKey())
+        .isLessThanOrEqualTo(keyFromString(storeKeyRawPrefix + (char) 1 + "baz"));
+  }
+
+  @Test
+  public void prefix() {
+    ContentKey bar = ContentKey.of("bar");
+    KeyRanges keyRange = keyRanges(null, null, null, bar);
+
+    soft.assertThat(keyRange.beginStoreKey()).isEqualTo(keyToStoreKeyPrefixMin(bar));
+
+    soft.assertThat(keyRange.endStoreKey()).isGreaterThan(keyToStoreKey(bar));
+    soft.assertThat(keyRange.endStoreKey()).isGreaterThan(keyToStoreKeyVariant(bar, "Z"));
+    soft.assertThat(keyRange.endStoreKey()).isGreaterThan(keyToStoreKeyVariant(bar, "A"));
+
+    soft.assertThat(keyRange.endStoreKey())
+        .isGreaterThan(keyToStoreKey(ContentKey.of("bar", "baz")));
+    soft.assertThat(keyRange.endStoreKey()).isGreaterThan(keyToStoreKey(ContentKey.of("bar", "0")));
+    soft.assertThat(keyRange.endStoreKey()).isGreaterThan(keyToStoreKey(ContentKey.of("barbaz")));
```

Comment:
This fails, though:
```
    soft.assertThat(keyRange.endStoreKey()).isGreaterThan(keyToStoreKey(ContentKey.of("bar\u03C0")));
```

Reply:
Yes - why should it not? It's `> "bar"`

Reply:
But it has the right prefix.... why `barbaz` passes by `bar\u03C0` fails?

Reply:
It's a negative `byte` value in that UTF char - and this one just ends in a string comparison.

Reply:
but we support Unicode at the `ContentKey` level, right?.. and as a `ContentKey` is has the right prefix, so I guess it should be included in the range :thinking:

Reply:
To put it another way, should we use `Character.MAX_VALUE` instead of 255 in `TypeMapping`?

Reply:
Actually, it belongs to this PR. Just sad that this change requires such a change :(